### PR TITLE
Port for WCH CH32V series (CH32V003, CH32V103, and CH32V203)

### PR DESCRIPTION
**Pull request by nosuz**
I have added port modules and examples for WCH CH32V serires MPU.
Examples programs can be built by `make` in examples/wch.

I am happy to be merged my work.

However, I have found the other Pull-Request [#223](https://github.com/ZigEmbeddedGroup/microzig/pull/223) which supports CH32V003 is proposed. Should I merge this Pull-Request int my branch?

Thank you,

### CODE REVIEW

File: examples/wch/ch32v003/Makefile
Comment (ikskuh) [inline examples/wch/ch32v003/Makefile:1]: Same as with the other Makefile

File: examples/wch/ch32v103/LICENSE
Comment (ikskuh) [inline examples/wch/ch32v103/LICENSE:1]: Please delete this file, we don't need a copyright per port

File: examples/wch/ch32v103/Makefile
Comment (ikskuh) [inline examples/wch/ch32v103/Makefile:1]: See other Makefile comments

File: examples/wch/ch32v203/LICENSE
Comment (ikskuh) [inline examples/wch/ch32v203/LICENSE:1]: Please delete this file, we don't need a copyright per port

File: examples/wch/ch32v203/Makefile
Comment (ikskuh) [inline examples/wch/ch32v203/Makefile:1]: See other Makefiles

File: port/wch/ch32v/LICENSE
Comment (ikskuh) [inline port/wch/ch32v/LICENSE:1]: Please delete this file, we don't need a copyright per port

File: port/wch/ch32v/src/chips/ch32v003.zig
Comment (ikskuh) [inline port/wch/ch32v/src/chips/ch32v003.zig:1]: Please do not check in the register files generated by regz. You can directly use the `svd` files as register definitions.

File: port/wch/ch32v/src/chips/ch32v103.zig
Comment (ikskuh) [inline port/wch/ch32v/src/chips/ch32v103.zig:1]: Please do not check in the register files generated by regz. You can directly use the `svd` files as register definitions.

File: port/wch/ch32v/src/chips/ch32v103xx.zig
Comment (ikskuh) [inline port/wch/ch32v/src/chips/ch32v103xx.zig:1]: Please do not check in the register files generated by regz. You can directly use the `svd` files as register definitions.

File: port/wch/ch32v/src/chips/ch32v20x.zig
Comment (ikskuh) [inline port/wch/ch32v/src/chips/ch32v20x.zig:1]: Please do not check in the register files generated by regz. You can directly use the `svd` files as register definitions.

**Review by ikskuh — CHANGES_REQUESTED** [review]
This review was just a quick structural review, but it looks like you've made big progress on WCH front, which is great!

Will take a closer look if i find the time

File: examples/wch/Makefile
[the remark this answers is not in the document]
Comment (nosuz) [reply examples/wch/Makefile:1]: Thank you for reviewing my Pull-Request.
I think I have followed your all suggestions.

I removed the Makefiles and moved all examples into ch32v directory.

File: examples/wch/ch32v003/LICENSE
[the remark this answers is not in the document]
Comment (nosuz) [reply examples/wch/ch32v003/LICENSE:1]: I have removed the LICENSE files from exsample directorys.

File: examples/wch/ch32v003/Makefile
Comment (nosuz) [reply examples/wch/ch32v003/Makefile:1]: I have removed the Makefile.

File: examples/wch/ch32v103/Makefile
Comment (nosuz) [reply examples/wch/ch32v103/Makefile:1]: I have removed the Makefile.

File: examples/wch/ch32v103/LICENSE
Comment (nosuz) [reply examples/wch/ch32v103/LICENSE:1]: I have removed the LICENSE file.

File: examples/wch/ch32v203/LICENSE
Comment (nosuz) [reply examples/wch/ch32v203/LICENSE:1]: I have removed the LICENSE file.

File: examples/wch/ch32v203/Makefile
Comment (nosuz) [reply examples/wch/ch32v203/Makefile:1]: I have removed the Makefile.

File: port/wch/ch32v/LICENSE
Comment (nosuz) [reply port/wch/ch32v/LICENSE:1]: I have removed the LICENSE file.

File: port/wch/ch32v/src/chips/ch32v103.zig
Comment (nosuz) [reply port/wch/ch32v/src/chips/ch32v103.zig:1]: I removed the register file generated by regz.

File: port/wch/ch32v/src/chips/ch32v103xx.zig
Comment (nosuz) [reply port/wch/ch32v/src/chips/ch32v103xx.zig:1]: I removed the register file generated by regz.

File: port/wch/ch32v/src/chips/ch32v20x.zig
Comment (nosuz) [reply port/wch/ch32v/src/chips/ch32v20x.zig:1]: I removed the register file generated by regz.

Thank you,

**Comment by nosuz** [comment]
Thank you for your kindly revew and sugestions.
I have followed your all suggestions and updated my branch to integrate the chages in your main branch.

**Comment by nosuz** [comment]
@mattnite Thank you for your advice. I have rebased to your current main branch. And I have also changed build.zig files in the top and examples directories to add "wch/ch32".

**Comment by mattnite** [comment]
@nosuz thank you!

**Review by mattnite — APPROVED** [review]

**Comment by nosuz** [comment]
@mattnite Thank you very much for merging my Pull-Request.

But I am sorry I found my examples fail to compile becase of the chnage by "Build system rewrite" [#259](https://github.com/ZigEmbeddedGroup/microzig/pull/259). I am going to fix.

Should I mention in this Pull-Request tree when I finished my work, or should I create new Pull-Request?

**Comment by mattnite** [comment]
@nosuz No worries! I realized this may be the case. There's a few other issues that need fixing before the next release. I wanted to get all your good work in now, and then fix it up later when I had a couple patterns figured out for CI.

**Comment by nosuz** [comment]
@mattnite Thank you for your advices. I am trying to fix this problem and will send a new PR when it is finished.

**Pull request merged**